includes/ziti/ziti_events.h

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright (c) 2020 NetFoundry, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+#ifndef ZITI_SDK_ZITI_EVENTS_H
+#define ZITI_SDK_ZITI_EVENTS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+    ZitiContextEvent = 1,
+    ZitiRouterEvent = 1 << 1,
+    ZitiServiceEvent = 1 << 2,
+} ziti_event_type;
+
+typedef enum {
+    EdgeRouterConnected,
+    EdgeRouterDisconnected,
+    EdgeRouterRemoved,
+    EdgeRouterUnavailable,
+} ziti_router_status;
+
+struct ziti_context_event {
+    int ctrl_status;
+    const char *err;
+};
+
+struct ziti_router_event {
+    ziti_router_status status;
+    const char* name;
+    const char* version;
+};
+
+struct ziti_service_event {
+    ziti_service_array removed;
+    ziti_service_array changed;
+    ziti_service_array added;
```

Comment:
should added events be here? or should those still come by `service_cb` from `init_opts`? seems like a double dip to me? or are events replacing the `service_cb` entirely? i'm not sure how these two are supposed to correlate

Reply:
service_cb is deprecated, for now both are supported